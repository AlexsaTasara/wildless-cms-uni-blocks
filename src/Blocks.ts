import type { PlaceholderContent } from './Placeholder';
import { Placeholder } from './Placeholder';
import type { TextContent } from './Text';
import { Text } from './Text';

export type BlockContent = PlaceholderContent | TextContent;

export const Blocks = {
  Text,
  Placeholder,
};
