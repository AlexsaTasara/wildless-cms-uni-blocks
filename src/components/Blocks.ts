import { Accordion } from './Accordion/Accordion';
import { BenefitsBlock } from './BenefitsBlock/BenefitsBlock';
import { Bonus } from './Bonus/Bonus';
import { ComparisonTable } from './ComparisonTable/ComparisonTable';
import { CreditCalculator } from './CreditCalculator/CreditCalculator';
import { ExchangeRateTile } from './ExchangeRateTile/ExchangeRateTile';
import { Footer } from './Footer/Footer';
import { Gallery } from './Gallery/Gallery';
import { GracePeriod } from './GracePeriod/GracePeriod';
import { Header } from './Header/Header';
import { Headline } from './Headline/Headline';
import { LinkDocs } from './LinkDocs/LinkDocs';
import { MiniGallery } from './MiniGallery/MiniGallery';
import { MobileAppTile } from './MobileAppTile/MobileAppTile';
import { Navigation } from './Navigation/Navigation';
import { NavigatorTabs } from './NavigatorTabs/NavigatorTabs';
import { OtherProducts } from './OtherProducts/OtherProducts';
import { PictureText } from './PictureText/PictureText';
import { Placeholder } from './Placeholder/Placeholder';
import { ProductBlock } from './ProductBlock/ProductBlock';
import { ProductGallery } from './ProductGallery/ProductGallery';
import { ProductGalleryGreen } from './ProductGalleryGreen/ProductGalleryGreen';
import { ProductTile } from './ProductTile/ProductTile';
import { PromoTile } from './PromoTile/PromoTile';
import { Recommendation } from './Recommendation/Recommendation';
import { StepsBlock } from './StepsBlock/StepsBlock';
import { TariffsTable } from './TariffsTable/TariffsTable';
import { TextBlock } from './TextBlock/TextBlock';
import { Tile } from './Tile/Tile';
import { GroupBlock } from './GroupBlock/GroupBlock';

export const Blocks = {
  Accordion,
  BenefitsBlock,
  Bonus,
  ComparisonTable,
  CreditCalculator,
  ExchangeRateTile,
  Footer,
  Gallery,
  GracePeriod,
  Header,
  Headline,
  LinkDocs,
  MiniGallery,
  MobileAppTile,
  NavigatorTabs,
  PictureText,
  ProductBlock,
  ProductGallery,
  ProductGalleryGreen,
  ProductTile,
  PromoTile,
  OtherProducts,
  Recommendation,
  StepsBlock,
  TariffsTable,
  TextBlock,
  Tile,
  Navigation,
  Placeholder,
  GroupBlock,
};
