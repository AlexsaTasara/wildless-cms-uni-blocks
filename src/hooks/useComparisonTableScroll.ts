export const useComparisonTableScroll = <ColData>({
  colData,
  colsLengthForScroll,
  activeCardIndex,
  setActiveCardIndex,
}: {
  colData: ColData[];
  colsLengthForScroll: number;
  activeCardIndex: number;
  setActiveCardIndex: (_: number) => void;
}) => {
  const onNextClick = () => setActiveCardIndex(activeCardIndex + 1);
  const onPrevClick = () => setActiveCardIndex(activeCardIndex - 1);

  const isScrollAvailable = Boolean(colData?.length && colData.length > colsLengthForScroll);
  const showNextButton =
    isScrollAvailable && colData?.length - activeCardIndex > colsLengthForScroll;
  const showPrevButton = isScrollAvailable && activeCardIndex > 0;

  return { onNextClick, onPrevClick, isScrollAvailable, showNextButton, showPrevButton };
};
