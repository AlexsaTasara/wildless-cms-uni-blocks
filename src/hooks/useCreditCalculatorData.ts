import type { AsyncDataHook } from '../components/ContentPage/ContentPageContext';

export function useCreditCalculatorData(
  useAsyncData: AsyncDataHook,
  directoryName: string | undefined,
): any {
  const { data } = useAsyncData(creditCalculatorUrl(directoryName), fetchCreditCalculatorData);
  return data || {};
}

async function fetchCreditCalculatorData(url: string): Promise<any> {
  const response = await fetch(url);
  return await response.json();
}

function creditCalculatorUrl(directoryName: string | undefined) {
  return `/wcms-resources/${directoryName || 'credit-calculator-data'}.json`;
}
