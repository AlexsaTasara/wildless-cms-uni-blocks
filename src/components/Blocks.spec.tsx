import { setupHooks } from '@redneckz/uni-jsx/lib/hooks';
import { setup } from '@redneckz/uni-jsx/lib/setup';
import { render } from '@testing-library/react';
import * as React from 'react';
import runtime from 'react/jsx-runtime';
import { Blocks } from './Blocks';
import type { ContentPageContext } from './ContentPage/ContentPageContext';

const { jsx, jsxs } = runtime as any;

setup(jsx, jsxs);
setupHooks(React as any);

const emptyFn = () => {
  /* For sure */
};

const context: ContentPageContext = {
  useEffect: React.useEffect,
  useRouter: () => ({
    pathname: '/credits',
    query: {},
    push: emptyFn,
    replace: emptyFn,
  }),
  useAsyncData: () => ({}),
  useGeolocation: (defaultLocation) => [defaultLocation, emptyFn],
  useLikeService: () => ({
    likeCount: 0,
    like: emptyFn,
    dislike: emptyFn,
  }),
  handlerDecorator: (): any => emptyFn,
  useSearch: () => ({
    term: '',
    setTerm: emptyFn,
  }),
  IntersectionObserverTag: ({ children }) => children,
};

describe('Blocks', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  for (const [key, BlockComponent] of Object.entries(Blocks)) {
    describe(key, () => {
      it(`should render block "${key}" without content`, () => {
        expect.assertions(2);

        const { container } = render(<BlockComponent context={context} />);
        expect(container.firstChild).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
      });

      it(`should render block "${key}" with demo content`, async () => {
        expect.assertions(1);

        let data = {};

        try {
          const { default: example } = await import(`./${key}/${key}.example.json`);
          data = example;
        } catch (ex) {
          // Do nothing
        }

        const { container } = render(<BlockComponent context={context} {...data} />);
        expect(container.firstChild).toBeTruthy();
      });
    });
  }
});
