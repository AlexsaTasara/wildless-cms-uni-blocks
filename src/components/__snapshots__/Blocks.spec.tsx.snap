// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blocks Accordion should render block "Accordion" without content 1`] = `
<section
  class="p-[50px] mb-1 last:mb-0 font-sans bg-white text-primary-text "
/>
`;

exports[`Blocks BenefitsBlock should render block "BenefitsBlock" without content 1`] = `
<section
  class="font-sans text-primary-text bg-white p-12 flex flex-col items-center "
/>
`;

exports[`Blocks Bonus should render block "Bonus" without content 1`] = `
<section
  class="font-sans bg-white p-[50px] pb-9 "
>
  <div
    class="container"
  >
    <section
      class="font-sans flex flex-col p-[50px] gap-3 bg-transparent !p-0"
    />
    <div
      class="flex flex-wrap w-full mt-9"
    />
  </div>
</section>
`;

exports[`Blocks BonusBenefitsBlock should render block "BonusBenefitsBlock" without content 1`] = `
<section
  class="font-sans text-primary-text bg-white p-50 flex flex-col text-center "
>
  <section
    class="font-sans flex flex-col p-[50px] gap-3 bg-transparent !p-0"
  />
</section>
`;

exports[`Blocks Calculator should render block "Calculator" without content 1`] = `
<section
  class="box-border overflow-hidden relative font-sans w-100 text-primary-text "
>
  <div
    class="bg-white"
  >
    <div />
  </div>
  <div
    class="flex"
    role="list"
    style="transform: translateX(-000%);"
  />
</section>
`;

exports[`Blocks Catalog should render block "Catalog" without content 1`] = `
<section
  class="bg-white relative font-sans p-12 overflow-hidden text-center "
>
  <section
    class="font-sans flex flex-col p-[50px] gap-3 bg-transparent !p-0"
  />
  <div
    class="flex duration-1000 gap-3.5 mt-8"
    role="list"
    style="transform: translateX(-0px);"
  />
  <div />
  <div
    class="absolute top-0 bottom-0 w-[84px] left-0 bg-opacity-from-white"
  />
  <div
    class="absolute top-0 bottom-0 w-[84px] right-0 bg-opacity-to-white"
  />
</section>
`;

exports[`Blocks ComparisonTable should render block "ComparisonTable" without content 1`] = `
<section
  class="bg-white font-sans py-[50px] pl-[50px] overflow-hidden text-primary-text relative "
>
  <h3
    class="font-sans m-0 text-h2 max-w-[47rem] text-center mb-9 mx-auto"
  />
  <div>
    <div
      role="table"
    />
  </div>
</section>
`;

exports[`Blocks ExchangeRateTile should render block "ExchangeRateTile" without content 1`] = `
<section
  class="bg-white text-primary-text font-sans p-9 box-border "
>
  <div
    class="font-sans flex flex-col grow h-full items-start "
  >
    <h3
      class="font-sans m-0 text-h4 whitespace-pre-wrap text-h4"
    >
      Курсы обмена валют
    </h3>
    <div
      class="flex grow w-full justify-between"
    >
      <div
        class="flex flex-col w-full"
      >
        <div>
          <div
            class="flex"
          >
            <div
              class="mr-14 pt-4"
            >
              <table
                class="h-fit w-fit mb-[30px]"
              >
                <thead>
                  <tr>
                    <th
                      class="text-left text-m-light text-secondary-text"
                    >
                      Валюта
                    </th>
                    <th
                      class="text-left text-m-light text-secondary-text pl-11"
                    >
                      Купить
                    </th>
                    <th
                      class="text-left text-m-light text-secondary-text pl-11"
                    >
                      Продать
                    </th>
                  </tr>
                </thead>
                <tbody />
              </table>
              <div
                class="flex "
              >
                <svg
                  aria-hidden="true"
                  class=""
                  height="24"
                  width="24"
                >
                  <title>
                    Иконка Gps
                  </title>
                  <desc>
                    Иконка Gps
                  </desc>
                  <use
                    class="text-primary-main "
                    href="icons/GpsIcon.svg#icon"
                    xlink:href="icons/GpsIcon.svg#icon"
                  />
                </svg>
                <div
                  class="ml-3"
                >
                  <p
                    class="text-primary-main  m-0 mb-1 text-l"
                  >
                    Москва, ул.Ленина д.28
                  </p>
                  <p
                    class="text-secondary-text m-0 text-m"
                  >
                    Курс указан для ближайшего офиса - 
                    6км
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Blocks Footer should render block "Footer" without content 1`] = `
<footer
  class="p-9 bg-white font-sans "
>
  <div
    class="flex items-stretch gap-[54px] pb-[30px] xl:gap-8"
  >
    <a
      aria-label="Россельхозбанк"
      class="inline-flex items-center font-sans no-underline w-[204px]"
      href="/"
      target="_self"
    >
      <svg
        aria-hidden="true"
        class="text-primary-main w-10"
        height="45"
        width="40"
      >
        <title>
          Иконка Logo
        </title>
        <desc>
          Иконка Logo
        </desc>
        <use
          class="text-primary-main "
          href="icons/Logo.svg#icon"
          xlink:href="icons/Logo.svg#icon"
        />
      </svg>
      <span
        class="text-black font-medium ml-2.5"
      >
        Россельхозбанк
      </span>
    </a>
    <form
      class="font-sans relative grow"
    >
      <div
        class="absolute rounded h-full flex items-center justify-center pl-4 max-w-[170px] gap-3.5 pointer-events-none"
      >
        <div
          class="w-[24px] h-[24px]"
        >
          <img
            alt="Иконка Лупа"
            aria-hidden="true"
            class=""
            height="24"
            src="icons/LoupeIcon.svg"
            title="Иконка Лупа"
            width="24"
          />
        </div>
        <label
          class="text-secondary-text text-l-light"
          for="search-bar-input"
        >
          Поиск по сайту
        </label>
      </div>
      <input
        class="h-12 pl-12 w-full peer text-l text-black border border-solid rounded-md box-border
              outline-none pr-[6%] border-main-divider focus:border-primary-text h-full"
        id="search-bar-input"
        name="search-bar-input"
        type="text"
        value=""
      />
      <a
        class="text-center font-sans select-none border border-transparent inline-block cursor-pointer no-underline focus:border-primary-focus focus:border text-white bg-primary-main hover:bg-primary-hover active:bg-primary-active rounded-md invisible peer-focus:visible absolute top-1 right-1"
        role="button"
      >
        <div
          class="text-s px-9 py-2.5"
        >
          Найти
        </div>
      </a>
    </form>
  </div>
  <div
    class="flex items-stretch gap-[54px] xl:gap-8"
  >
    <div
      class="w-[204px] flex flex-col shrink-0 overflow-hidden"
    >
      <div
        class=""
      >
        <a
          aria-label="Обратная связь"
          class="text-center font-sans select-none border border-transparent inline-block cursor-pointer no-underline focus:border-primary-focus focus:border text-white bg-primary-main hover:bg-primary-hover active:bg-primary-active rounded-md mb-6 w-full"
          href="/"
          role="link"
          target="_blank"
        >
          <div
            class="flex items-center justify-center gap-2 py-4 px-9 "
          >
            <div>
              <div
                class="text-xs-light text-left"
              />
              <div
                class="text-left text-l"
              >
                Обратная связь
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        class="font-sans"
      >
        <div
          class="flex gap-2.5 "
        />
        <span
          class="text-s-light text-secondary-text"
        >
          Соцсети
        </span>
      </div>
      <div
        class="font-sans"
      >
        <div
          class="flex gap-2.5 mt-6"
        />
        <span
          class="text-s-light text-secondary-text"
        >
          Мобильное приложение
        </span>
      </div>
    </div>
    <div
      class="flex flex-grow items-start justify-between gap-[54px] xl:gap-5 pt-[3px]"
    />
  </div>
  <div
    class="mt-[98px]"
  />
  <div
    class=""
  >
    <div
      class="pt-5"
    >
      <span
        class="text-xs-light text-secondary-text"
      >
        © 2000-2022 АО «Россельхозбанк» Генеральная лицензия Банка России № 3349 от 12.08.2015 г.
      </span>
    </div>
  </div>
</footer>
`;

exports[`Blocks Gallery should render block "Gallery" without content 1`] = `
<section
  class="relative font-sans text-primary-text bg-white p-[42px] overflow-hidden "
>
  <div>
    <div
      class="absolute top-0 left-0 bottom-0 w-[84px] bg-gradient-to-r from-white to-transparent"
    />
    <section
      class="font-sans flex flex-col p-[50px] gap-4 bg-transparent !p-0"
    />
    <div
      class="flex mt-8 justify-center duration-1000"
      role="list"
      style="transform: translateX(-0px);"
    />
  </div>
  <div />
  <div
    class="absolute top-0 left-0 bottom-0 w-[82px] bg-opacity-from-white"
  />
  <div
    class="absolute top-0 right-0 bottom-0 w-[82px] bg-opacity-to-white"
  />
</section>
`;

exports[`Blocks GracePeriod should render block "GracePeriod" without content 1`] = `
<section
  class="font-sans bg-white p-[50px] "
>
  <section
    class="font-sans flex flex-col p-[50px] gap-3 bg-transparent !p-0"
  />
</section>
`;

exports[`Blocks GroupBlock should render block "GroupBlock" without content 1`] = `
<section
  class="box-border "
/>
`;

exports[`Blocks Header should render block "Header" without content 1`] = `
<header
  class="pt-6 pb-5 px-20 bg-white "
>
  <div
    class="container"
  >
    <div
      class="flex items-center"
    >
      <a
        aria-label="Россельхозбанк"
        class="inline-flex items-center font-sans no-underline mr-8"
        href="/"
        target="_self"
      >
        <svg
          aria-hidden="true"
          class="text-primary-main w-10"
          height="45"
          width="40"
        >
          <title>
            Иконка Logo
          </title>
          <desc>
            Иконка Logo
          </desc>
          <use
            class="text-primary-main "
            href="icons/Logo.svg#icon"
            xlink:href="icons/Logo.svg#icon"
          />
        </svg>
        <span
          class="text-black font-medium ml-2.5"
        >
          Россельхозбанк
        </span>
      </a>
      <div
        class="flex items-center ml-auto"
      >
        <a
          aria-label="Местоположение"
          class="border-transparent inline-block border border-solid bg-transparent text-center no-underline  mr-5"
          href="#"
          rel="noopener noreferrer"
        >
          <span
            class="text-primary-text hover:text-primary-main font-sans text-s-light"
          >
            Москва
          </span>
        </a>
        <a
          aria-label="Список всех доступных офисов и банкоматов"
          class="border-transparent inline-block border border-solid bg-transparent text-center no-underline  mr-7"
          href="#"
          rel="noopener noreferrer"
        >
          <span
            class="text-primary-text hover:text-primary-main font-sans text-s-light"
          >
            Офисы и банкоматы
          </span>
        </a>
        <button
          aria-label="Поиск по сайту"
          class="border-0 p-0 w-[24px] h-[24px] flex items-center bg-inherit cursor-pointer mr-5 group text-primary-text hover:text-primary-main"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="h-full pointer-events-none"
          >
            <title>
              Иконка Лупа
            </title>
            <desc>
              Иконка Лупа
            </desc>
            <use
              class="text-black group-hover:text-primary-main pointer-events-none"
              href="icons/LoupeIcon.svg#icon"
              xlink:href="icons/LoupeIcon.svg#icon"
            />
          </svg>
        </button>
        <button
          aria-label="Дополнительные ссылки"
          class="border-0 p-0 w-[24px] h-[24px] flex items-center bg-inherit cursor-pointer text-secondary-light hover:text-secondary-hover group min-w-[32px] min-h-[32px]"
          disabled=""
          type="button"
        >
          <svg
            aria-hidden="true"
            class="h-full bg-main-divider rounded-full pointer-events-none"
          >
            <title>
              Иконка Сетка (белая)
            </title>
            <desc>
              Иконка Сетка (белая)
            </desc>
            <use
              class="text-black group-hover:text-primary-main"
              href="icons/GridIcon.svg#icon"
              xlink:href="icons/GridIcon.svg#icon"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="mt-6 h-[1px] bg-main-divider"
    />
  </div>
</header>
`;

exports[`Blocks Headline should render block "Headline" without content 1`] = `
<section
  class="font-sans flex flex-col p-[50px] gap-4 bg-transparent "
/>
`;

exports[`Blocks InsuranceAmountBlock should render block "InsuranceAmountBlock" without content 1`] = `
<section
  class="box-border py-[50px] overflow-hidden relative font-sans w-100 bg-white "
>
  <div
    class="flex"
    role="list"
    style="transform: translateX(-000%);"
  />
</section>
`;

exports[`Blocks LinkDocs should render block "LinkDocs" without content 1`] = `
<section
  class="font-sans text-primary-text p-[50px] bg-white "
>
  <section
    class="font-sans flex flex-col p-[50px] gap-3 bg-transparent !p-0"
  />
</section>
`;

exports[`Blocks MiniGallery should render block "MiniGallery" without content 1`] = `
<section
  class="relative font-sans text-primary-text bg-white p-[42px] overflow-hidden "
>
  <div>
    <div
      class="absolute top-0 left-0 bottom-0 w-[84px] bg-gradient-to-r from-white to-transparent"
    />
    <section
      class="font-sans flex flex-col p-[50px] gap-4 bg-transparent !p-0"
    />
    <div
      class="flex mt-8 justify-center duration-1000"
      role="list"
      style="transform: translateX(-0px);"
    />
  </div>
  <div />
  <div
    class="absolute top-0 left-0 bottom-0 w-[82px] bg-opacity-from-white"
  />
  <div
    class="absolute top-0 right-0 bottom-0 w-[82px] bg-opacity-to-white"
  />
</section>
`;

exports[`Blocks MobileAppTile should render block "MobileAppTile" without content 1`] = `
<section
  class="flex justify-between font-sans p-9 box-border relative min-h-[320px]
        min-h-[360px]
        bg-white text-primary-text
        "
>
  <div
    class="font-sans flex flex-col grow h-full items-start z-[1]"
  >
    <h3
      class="font-sans m-0 text-h4 whitespace-pre-wrap max-w-[509px] mb-6
                  text-primary-text"
    >
      Мобильное приложение
    </h3>
    <div
      class="flex grow w-full justify-between"
    >
      <div
        class="flex flex-col w-full"
      >
        <div>
          <div
            class="flex flex-1 items-center"
          >
            <span
              class="text-m-light text-secondary-text"
            >
              Наведите камеру телефона на QR-код и скачайте приложение
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Blocks Navigation should render block "Navigation" without content 1`] = `
<section
  class="font-sans text-primary-text rounded-[40px] flex items-center justify-between "
>
  <h2
    class="text-xl font-medium m-0"
  />
</section>
`;

exports[`Blocks NavigatorTabs should render block "NavigatorTabs" without content 1`] = `
<section
  class="sticky top-0 z-20 box-border flex gap-x-1 h-14 "
/>
`;

exports[`Blocks OfficeCard should render block "OfficeCard" without content 1`] = `
<section
  class="font-sans text-primary-text py-10 bg-white "
  id=""
>
  <section />
</section>
`;

exports[`Blocks OtherProducts should render block "OtherProducts" without content 1`] = `
<section
  class="font-sans text-primary-text "
/>
`;

exports[`Blocks PictureText should render block "PictureText" without content 1`] = `
<section
  class="relative font-sans p-14 bg-white text-primary-text "
>
  <section
    class="font-sans flex flex-col p-[50px] gap-3 bg-white !p-0"
  />
  <div
    class="flex justify-center gap-[122px] mt-[30px]"
  >
    <div
      class="w-[558px] m-auto"
    />
  </div>
</section>
`;

exports[`Blocks Placeholder should render block "Placeholder" without content 1`] = `
<section
  class="bg-white text-primary-text font-sans p-9 box-border undefined pr-[9.4rem] min-h-[360px] "
>
  <section
    class="font-sans flex flex-col p-[50px] gap-4 bg-transparent !p-0 max-w-[600px]"
  >
    <div
      class="text-xl-light text-primary-text text-center"
    >
      Блок в разработке...
    </div>
  </section>
  <figure
    class="m-0 min-w-[80%]"
  >
    <div
      class="h-4 mt-2 bg-secondary-light"
      style="width: 100%;"
    />
    <div
      class="h-4 mt-2 bg-secondary-light"
      style="width: 61%;"
    />
    <div
      class="h-4 mt-2 bg-secondary-light"
      style="width: 38%;"
    />
    <div
      class="h-4 mt-2 bg-secondary-light"
      style="width: 23%;"
    />
  </figure>
</section>
`;

exports[`Blocks ProductBlock should render block "ProductBlock" without content 1`] = `
<section
  class="font-sans overflow-hidden p-[50px] flex flex-col box-border min-h-[528px] bg-white text-primary-text "
>
  <div
    class="flex grow justify-between items-stretch h-full "
  >
    <div
      class="flex flex-col "
    >
      <section
        class="font-sans flex flex-col p-[50px] gap-4 bg-white !p-0 max-w-[600px]"
      />
      <div
        class="font-sans flex flex-col grow h-full items-start "
      >
        
        <div
          class="flex grow w-full justify-between"
        >
          <div
            class="flex flex-col w-full"
          >
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Blocks ProductGallery should render block "ProductGallery" without content 1`] = `
<section
  class="box-border pt-[50px] overflow-hidden relative font-sans w-100 
        bg-white text-primary-text"
>
  <div
    class="flex duration-1000 pb-14"
    role="list"
    style="transform: translateX(-000%);"
  />
  <div
    class="flex items-center absolute bottom-6 left-0 right-0 px-[26px] box-border h-28"
  />
</section>
`;

exports[`Blocks ProductTile should render block "ProductTile" without content 1`] = `
<section
  class="bg-white overflow-hidden text-primary-text font-sans p-9 box-border min-h-[364px] relative justify-between grid  bg-white text-primary-text pr-[9.4rem] min-h-[360px] "
>
  <div
    class="z-[1]"
  >
    <section
      class="font-sans flex flex-col p-[50px] gap-2 bg-white !p-0"
    />
    <div
      class="flex mt-3.5"
    />
  </div>
</section>
`;

exports[`Blocks PromoTile should render block "PromoTile" without content 1`] = `
<section
  class="bg-white text-primary-text font-sans p-9 box-border  bg-white text-primary-text pr-[9.4rem] min-h-[360px] "
>
  <div
    class="font-sans flex flex-col grow h-full items-start "
  >
    <div>
      <section
        class="font-sans flex flex-col p-[50px] gap-4 bg-white !p-0 max-w-[600px]"
      >
        <h2
          class="font-sans m-0 text-h2 whitespace-pre-wrap text-primary-text text-left"
        >
          Акции и спецпредложения
        </h2>
      </section>
    </div>
    <div
      class="flex grow w-full justify-between"
    >
      <div
        class="flex flex-col w-full"
      >
        <div>
          <div
            class="mt-1.5"
          >
            <div
              class="text-l"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Blocks Recommendation should render block "Recommendation" without content 1`] = `
<section
  class="relative font-sans p-9 overflow-hidden text-center bg-white text-primary-text "
>
  <div>
    <div
      class="flex duration-1000 gap-3.5"
      role="list"
      style="transform: translateX(-0px);"
    />
    <div>
      <div />
    </div>
  </div>
  <div
    class="absolute top-0 bottom-0 w-[84px] left-0 bg-opacity-from-white"
  />
  <div
    class="absolute top-0 bottom-0 w-[84px] right-0 bg-opacity-to-white"
  />
</section>
`;

exports[`Blocks SafeDepositRental should render block "SafeDepositRental" without content 1`] = `
<section
  class="bg-white px-8 py-12 "
>
  <div
    class="flex justify-between align-top mb-6"
  >
    <div
      class="flex-1 mr-9"
    >
      <div
        class="flex justify-between mb-6"
      >
        <div
          class="w-full mr-4"
        >
          <div>
            <span
              class="block text-m-light mb-2"
            >
              Город
            </span>
            <select
              class="appearance-none h-14 border border-gray p-4 rounded-md text-s w-full cursor-pointer"
              style="background: url(/icons/ArrowDownIcon.svg) no-repeat 50%;"
            >
              <option
                class=""
                value="Москва"
              >
                Москва
              </option>
              <option
                class=""
                value="Краснодар"
              >
                Краснодар
              </option>
            </select>
          </div>
        </div>
        <div
          class="w-full"
        >
          <div>
            <span
              class="block text-m-light mb-2"
            >
              Отделение
            </span>
            <select
              class="appearance-none h-14 border border-gray p-4 rounded-md text-s w-full cursor-pointer"
              style="background: url(/icons/ArrowDownIcon.svg) no-repeat 50%;"
            >
              <option
                class=""
                value="МоскваОфис1"
              >
                МоскваОфис1
              </option>
              <option
                class=""
                value="МоскваОфис2"
              >
                МоскваОфис2
              </option>
              <option
                class=""
                value="МоскваОфис3"
              >
                МоскваОфис3
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        class="mb-6"
      >
        <span
          class="block text-m-light mb-[3px]"
        >
          Срок аренды
        </span>
        <div>
          <label
            class="block relative"
          >
            <input
              class="m-0 mt-2 font-sans text-h3 w-full h-14 border border-solid border-main-stroke rounded-md
                        outline-none p-0 pl-4 box-border text-primary-text hover:border-primary-hover focus:border-primary-focus disabled:border-main-stroke disabled:bg-main-divider"
              value="26"
            />
            <div
              class="absolute inset-x-0 -bottom-1 px-4 leading-[18px]"
            >
              <input
                class="box-border w-full m-0 cursor-pointer slider"
                max="365"
                min="1"
                step="1"
                style="background-size: 6.87% 100%;"
                type="range"
                value="26"
              />
            </div>
          </label>
          <div
            class="flex justify-between my-2"
          >
            <span
              class="text-s-light text-secondary-text"
            >
              От 1 дня
            </span>
            <span
              class="text-s-light text-secondary-text"
            >
              До 365 дней
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex justify-between mb-6"
      >
        <div
          class="mr-4 w-full"
        >
          <div>
            <span
              class="block text-m-light mb-2"
            >
              Размер ячейки
            </span>
            <select
              class="appearance-none h-14 border border-gray p-4 rounded-md text-s w-full cursor-pointer"
              style="background: url(/icons/ArrowDownIcon.svg) no-repeat 50%;"
            >
              <option
                class=""
                value="маленькая"
              >
                маленькая
              </option>
              <option
                class=""
                value="средняя"
              >
                средняя
              </option>
              <option
                class=""
                value="большая"
              >
                большая
              </option>
            </select>
          </div>
        </div>
        <div
          class="w-full"
        >
          <div>
            <span
              class="block text-m-light mb-2"
            >
              Параметры ячейки, мм (ВхШхД)
            </span>
            <select
              class="appearance-none h-14 border border-gray p-4 rounded-md text-s w-full cursor-pointer"
              style="background: url(/icons/ArrowDownIcon.svg) no-repeat 50%;"
            >
              <option
                class=""
                value="125x225x375"
              >
                125x225x375
              </option>
              <option
                class=""
                value="200x300x400"
              >
                200x300x400
              </option>
              <option
                class=""
                value="300x400x500"
              >
                300x400x500
              </option>
            </select>
          </div>
        </div>
      </div>
      <span
        class="text-secondary-text text-l-light"
      >
        Тип договора
      </span>
      <div
        class="flex justify-between items-center mt-5"
      >
        <div>
          <div
            class="mb-2"
          >
            <label
              class="flex items-center relative cursor-pointer"
            >
              <input
                checked=""
                class="appearance-none flex justify-center items-center w-5 h-5 border-2 border-primary-main rounded-full cursor-pointer after:block after:w-2.5 after:h-2.5 after:rounded-full checked:after:bg-primary-main"
                name="contractType"
                type="radio"
                value="option1"
              />
              <span
                class="font-sans ml-3 cursor-pointer"
              >
                Простое хранение
              </span>
            </label>
          </div>
          <div>
            <label
              class="flex items-center relative cursor-pointer"
            >
              <input
                class="appearance-none flex justify-center items-center w-5 h-5 border-2 border-primary-main rounded-full cursor-pointer after:block after:w-2.5 after:h-2.5 after:rounded-full checked:after:bg-primary-main"
                name="contractType"
                type="radio"
                value="option2"
              />
              <span
                class="font-sans ml-3 cursor-pointer"
              >
                Сделка с недвижимостью
              </span>
            </label>
          </div>
        </div>
        <a
          class="text-center font-sans select-none border border-transparent inline-block cursor-pointer no-underline focus:border-primary-focus focus:border text-white bg-primary-main hover:bg-primary-hover active:bg-primary-active rounded-md "
          role="button"
        >
          <div
            class="flex items-center justify-center gap-2 py-4 px-9 "
          >
            <div>
              <div
                class="text-xs-light text-left"
              />
              <div
                class="text-left text-l"
              >
                Офисы на карте
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      class="flex flex-col justify-between"
    >
      <div
        class="flex"
      >
        <div
          class="mr-11"
        >
          <span
            class="block text-m-light mb-1"
          >
            Аренда за
          </span>
          <span
            class="text-h3"
          >
            26
          </span>
        </div>
        <div
          class=""
        >
          <span
            class="block text-m-light mb-1"
          >
            Цена
          </span>
          <span
            class="text-h3"
          >
            26 ₽
          </span>
        </div>
      </div>
      <div
        class="bg-primary-main px-10 pt-10 pb-4 text-center"
      >
        <svg
          aria-hidden="true"
          class=""
          height="257"
          width="290"
        >
          <title>
            Иконка PackA4
          </title>
          <desc>
            Иконка PackA4
          </desc>
          <use
            class="text-primary-main "
            href="icons/PackA4.svg#icon"
            xlink:href="icons/PackA4.svg#icon"
          />
        </svg>
        <span
          class="text-s-light text-white opacity-80 mt-2 inline-block"
        >
          На рисунки изображена пачка
          <br />
           листов формата А4
        </span>
      </div>
    </div>
  </div>
  <p
    class="text-s-light text-secondary-text"
  >
    Аренда СЯ, в случае проведения процедуры купли-продажи, с использованием наличных денег или процедуры ипотечной сделки
  </p>
</section>
`;

exports[`Blocks StepsBlock should render block "StepsBlock" without content 1`] = `
<section
  class="box-border font-sans text-primary-text bg-white px-[70px] py-[50px] flex flex-col items-center bg-white text-primary-text "
/>
`;

exports[`Blocks Tabs should render block "Tabs" without content 1`] = `
<section
  class="mb-2 box-border flex gap-x-1 undefined"
/>
`;

exports[`Blocks TariffsTable should render block "TariffsTable" without content 1`] = `
<section
  class="bg-white font-sans py-[50px] pl-[50px] overflow-hidden text-primary-text relative "
/>
`;

exports[`Blocks TextBlock should render block "TextBlock" without content 1`] = `
<section
  class="font-sans px-9 py-4 flex bg-white "
>
  <div
    class="py-0.5"
  />
</section>
`;

exports[`Blocks Tile should render block "Tile" without content 1`] = `
<section
  class="overflow-hidden font-sans p-9 pr-3 box-border  bg-white text-primary-text pr-[9.4rem] min-h-[360px] "
>
  <div
    class="font-sans flex flex-col grow h-full items-start "
  >
    <div
      class="flex grow w-full justify-between"
    >
      <div
        class="flex flex-col w-full"
      >
        <div />
      </div>
    </div>
  </div>
</section>
`;
